lib/iris/__init__.py

```diff
@@ -445,7 +453,10 @@ def sample_data_path(*path_to_join):
                          'NB. This function is only for locating files in the '
                          'iris sample data collection. It is not needed or '
                          'appropriate for general file access.'.format(target))
-    target = os.path.join(iris.config.SAMPLE_DATA_DIR, target)
+    if _SAMPLE_DATA_AVAILABLE:
+        target = os.path.join(iris_sample_data.path, target)
+    else:
+        target = os.path.join(iris.config.SAMPLE_DATA_DIR, target)
```

Comment:
I think there should be something like a "deprecation" of the config entry, to remind people to get rid of it.
It can be flagged up by the iris.config module when seen in the file, and also when ```iris.config.SAMPLE_DATA_DIR``` is accessed.